src/requests.jl

```diff
@@ -63,23 +63,23 @@ end
 end
 
 @inline function _generate_full_url(client::Client, path::AbstractString)::HTTP.URI
-    base_url = get_base_url(client)
-    result = _generate_full_url(base_url, path)
-    return result
-end
+    # Add trailing slash to base URL
+    # This is important since
+    # URIs.resolvereference("http://foo/bar", "./baz") = URIs.URI("http://foo/baz")
+    # whereas
+    # URIs.resolvereference("http://foo/bar/", "./baz") = URIs.URI("http://foo/bar/baz")
+    # (compliant with RFC 3986 Section 5.2)
+    base_url = _add_trailing_slash(_get_http_uri(get_base_url(client)))
 
-@inline function _generate_full_url(base_url::BaseURL, path::AbstractString)::HTTP.URI
-    base_url_uri = _get_http_uri(base_url)
-    result = _generate_full_url(base_url_uri, path)
-    return result
-end
+    # Treat all paths without scheme as relative
+    # Adding the dot is important since
+    # URIs.resolvereference("http://foo/bar/", "/baz") = URIs.URI("http://foo/baz")
+    # whereas
+    # URIs.resolvereference("http://foo/bar/", "./baz") = URIs.URI("http://foo/bar/baz")
+    # (compliant with RFC 3986 Section 5.2)
+    _path = replace(path, r"^/+" => "./")
```

Comment:
Is a regex necessary here? I think I'd find it easier to read as:

```
_path = startswith(path, "/") ? "." * path : path
```

Reply:
```suggestion
    _path = startswith(path, "/") ? "." * path : path
```

Reply:
If there's no actual downside to having the extra slashes, I think I prefer the non-regex approach, because it makes the code easier to read (IMO).